clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -279,11 +279,10 @@ static void DefineFastIntType(unsigned TypeWidth, bool IsSigned,
 /// Get the value the ATOMIC_*_LOCK_FREE macro should have for a type with
 /// the specified properties.
 static const char *getLockFreeValue(unsigned TypeWidth, unsigned TypeAlign,
-                                    unsigned InlineWidth) {
+                                    const TargetInfo &TI, bool IsCheriCap) {
   // Fully-aligned, power-of-2 sizes no larger than the inline
   // width will be inlined as lock-free operations.
-  if (TypeWidth == TypeAlign && (TypeWidth & (TypeWidth - 1)) == 0 &&
-      TypeWidth <= InlineWidth)
+  if (TI.hasBuiltinAtomic(TypeWidth, TypeAlign, IsCheriCap))
```

Comment:
I think this is fine in that the subtle differences won't ever matter here... might be worth trying to upstream that change though in case it's not?

Reply:
https://reviews.llvm.org/D135142

Reply:
Updated to cherry-pick that commit.